testdata/nil-len.go

```diff
@@ -35,4 +35,6 @@ func fn() {
 
 	if s != nil && len(s) == 0 { // nil check is not redundant here
 	}
+	if s != nil && len(s) != 1 { // nil check is not redundant here
```

Comment:
Yeah, it's true. Only `s != nil && len(s) != 0`and `s == nil || len(s) == 0' are redundant.